extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/remote_inference_client/conversational.py

```diff
@@ -0,0 +1,291 @@
+import copy
+import json
+from typing import TYPE_CHECKING, Any, List, Optional
+
+# HuggingFace API imports
+from huggingface_hub import InferenceClient
+
+from aiconfig import CallbackEvent
+from aiconfig.default_parsers.parameterized_model_parser import (
+    ParameterizedModelParser,
+)
+from aiconfig.model_parser import InferenceOptions
+from aiconfig.schema import (
+    ExecuteResult,
+    Output,
+    Prompt,
+    PromptMetadata,
+)
+from aiconfig.util.config_utils import get_api_key_from_environment
+from aiconfig.util.params import resolve_prompt
+
+
+# Circuluar Dependency Type Hints
+if TYPE_CHECKING:
+    from aiconfig.Config import AIConfigRuntime
+
+
+# Step 1: define Helpers
+def refine_completion_params(model_settings: dict[Any, Any]) -> dict[str, Any]:
+    """
+    Refines the completion params for the HF text translation api. Removes any unsupported params.
+    See https://github.com/huggingface/huggingface_hub/blob/main/src/huggingface_hub/inference/_client.py#L1703
+    for supported params. Note that client params DO NOT match API docs here:
+    https://huggingface.co/docs/api-inference/detailed_parameters#translation-task
+    """
+
+    supported_keys = {
+        "model",
+        "src_lang",
+        "tgt_lang",
+    }
+
+    completion_data = {}
+    for key in model_settings:
+        if key.lower() in supported_keys:
+            completion_data[key.lower()] = model_settings[key]
+
+    return completion_data
+
+
+def construct_output(response: str) -> Output:
+    metadata: dict[str, str] = {"raw_response": response}
+
+    output = ExecuteResult(
+        **{
+            "output_type": "execute_result",
+            "data": response,
+            "execution_count": 0,
+            "metadata": metadata,
+        }
+    )
+    return output
+
+
+class HuggingFaceTextTranslationRemoteInference(ParameterizedModelParser):
```

Comment:
I know we plan to change later, but just feel it's good to rename this in this intermediate diff anyways to ensure each intermediate step wont break. Not a blocker